storm-server/src/main/java/org/apache/storm/localizer/LocallyCachedBlob.java

```diff
@@ -162,6 +152,31 @@ protected static long getSizeOnDisk(Path p) throws IOException {
      */
     public abstract long getSizeOnDisk();
 
+    /**
+     * Get the size of p in bytes.
+     * @param p the path to read.
+     * @return the size of p in bytes.
+     */
+    protected static long getSizeOnDisk(Path p) throws IOException {
+        if (!Files.exists(p)) {
+            return 0;
+        } else if (Files.isRegularFile(p)) {
+            return Files.size(p);
+        } else {
+            //We will not follow sym links
+            return Files.walk(p)
+                    .filter((subp) -> Files.isRegularFile(subp, LinkOption.NOFOLLOW_LINKS))
+                    .mapToLong((subp) -> {
+                        try {
+                            return Files.size(subp);
+                        } catch (IOException e) {
+                            LOG.warn("Could not get the size of ");
```

Comment:
This does not look like it is complete.  Could not get the size of what?

Reply:
I merely moved this code up according to the checkstyle. I didn't make any change to the code itself.

Reply:
That may be, but if you could fix it that would be great.